tests/test_bfh.py

```diff
@@ -325,30 +354,96 @@ class Consistent(Mapping):
 
     def test_empty_fields_serialize_as_none_valid_or_no(self):
         """We aren't making assumptions here. Call validate if you want it."""
-        class FirstSchema(Schema):
-            wow = IntegerField(required=True)
-            umm = IntegerField(required=False)
+        first_schema, other_schema, my_map = set_up_schemas()
+        source = first_schema(wow=1)
+        assert source.validate()
 
-        class OtherSchema(Schema):
-            cool = IntegerField(required=True)
-            bad = IntegerField(required=True)  # Uh oh
+        transformed = my_map().apply(source)
+        with self.assertRaises(Invalid):
+            transformed.validate()
 
-        class Mymap(Mapping):
-            source_schema = FirstSchema
-            target_schema = OtherSchema
+        self.assertEqual({"cool": 1}, transformed.serialize())
 
-            cool = Get('wow')
-            bad = Get('umm')
+    def test_empty_fields_serialized_with_defaults(self):
+        first_schema, other_schema, my_map = set_up_schemas(2)
 
-        source = FirstSchema(wow=1)
+        source = first_schema(wow=1, umm=None)
         assert source.validate()
 
-        transformed = Mymap().apply(source)
-        with self.assertRaises(Invalid):
-            transformed.validate()
+        transformed = my_map().apply(source)
+
+        self.assertEqual({"cool": 1, "bad": 2}, transformed.serialize())
+
+#       test that explicit None is different from field not being present.
```

Comment:
I'll go even a little stronger here; this isn't just an opinion (that missing and None mean the same thing) -- one could argue either way on that-- it's a design choice that i'm still happy with. Show the keys or not show the keys on serialization, but they always mean the same. Please roll with this.